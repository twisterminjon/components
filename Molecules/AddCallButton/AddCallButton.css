.addcallbutton-wrap {
  position: relative;
}

.addcallbutton-button {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.addcallbutton-menu {
  position: absolute;
  background-color: var(--white);
  border-radius: 4px;
  margin-right: 20px;

  top: 12px;
  right: 58px;
  box-shadow: 0 2px 3px 0 rgba(34, 36, 38, 0.15);

  z-index: 1000;
}

.addcallbutton-menu:after {
  content: '';
  position: absolute;

  height: 0;
  width: 0;
  top: 10px;
  right: -16px;
  border: 8px solid transparent;
  border-left-color: var(--white);
}

.addcallbutton-menu--hide {
  display: none;
}

.addcallbutton-menu--show {
  display: static;
}

.addcallbutton-option {
  padding: 8px 16px;
  border-radius: 4px;
  white-space: nowrap;
}

.addcallbutton-option:hover {
  background-color: #eee;
}

.addcallbutton-option-label {
  margin-left: 8px;
  pointer-events: none;
  font-weight: 500;
}

/* ******************************* */
/* Menu */
/* ******************************* */

.addcallbuttonmenu-wrap {
  /* This required to handle the height animation, w/o the padding on the content will show  */
  position: fixed;
  width: 375px;
  height: 0;
  overflow-y: hidden;

  top: 0;
  left: 50%;
  margin-left: -179px;

  transition: height 200ms ease-out;
  z-index: var(--z-modal);
}

.addcallbuttonmenu-dimmer {
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  z-index: calc(var(--z-modal) - 1);
}

.addcallbuttonmenu {
  position: relative;
  color: white;

  width: 375px;
  padding: 16px;

  background: var(--brandcolor_verydark);

  border-bottom-left-radius: 38px;
  border-bottom-right-radius: 38px;
}

.addcallbuttonmenu--show {
  height: 429px;
  transition: height 200ms ease-in;
}

.addcallbuttonmenu h3 {
  color: var(--white);
  font-size: 20px;
  font-weight: 300;
  margin: 44px 0 16px 0;

  text-align: center;
}

.addcallbuttonmenu-buttons {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 10px;
  grid-row-gap: 10px;

  margin: 48px 0 0;
}

.addcallbuttonmenu button:first-of-type {
  margin-right: 10px;
}

.addcallbuttonmenu-back--row {
  display: flex;
  justify-content: center;
  align-items: center;
}

.addcallbuttonmenu-back {
  display: flex;
  align-items: center;

  margin-top: 46px;
  padding: 6px 8px;
  background: transparent;
  color: var(--brandcolor);

  font-size: 23px;

  border: none;
  outline: none;

  cursor: pointer;
}

.addcallbuttonmenu-back:hover {
  box-shadow: 0 0 4px 2px var(--button_color_pressed);
}

.addcallbuttonmenu-back:focus {
  box-shadow: 0 0 4px 2px var(--button_color_pressed);
}

/* Override semantic styles */
.addcallbuttonmenu-button {
  height: 85px;
  width: 165px;
  margin: 0 !important;

  color: var(--black) !important;
  background: var(--brandcolor) !important;
}

.addcallbuttonmenu-button:hover {
  background: var(--button_color_hover) !important;
}

.addcallbuttonmenu-button:focus {
  box-shadow: 0 0 4px 2px var(--button_color_pressed) !important;
}

.addcallbuttonmenu-button:active {
  background: var(--button_color_pressed) !important;
}

.ui.inverted.button.addcallbuttonmenu-button {
  color: var(--brandcolor) !important;
  background: none !important;
  box-shadow: 0 0 0 2px var(--brandcolor) inset !important;
}

.ui.inverted.button.addcallbuttonmenu-button:hover {
  background: var(--brandcolor_trans_25) !important;
}

.ui.inverted.button.addcallbuttonmenu-button:focus {
  background: var(--brandcolor_trans_25) !important;
}

.ui.inverted.button.addcallbuttonmenu-button:hover:active {
  background: var(--brandcolor_trans_50) !important;
}

/** Modal showing alert */

.modalalert h3 {
  height: 150px;
  padding: 30px 30px 10px 30px;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
}
