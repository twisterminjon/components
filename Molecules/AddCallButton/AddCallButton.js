import React, { Component } from 'react'
import PropTypes from 'prop-types'

import IconPlus from '../../Atoms/Icons/IconPlus'
import CallButtonLabel from '../../Atoms/CallButtonLabel/CallButtonLabel'
import CallButton from '../../Atoms/CallButton/CallButton'
import AddCallMenu from '../../Organisms/AddCallMenu/AddCallMenu'
import ModalToast from '../../Atoms/ModalToast/ModalToast'

import './AddCallButton.css'
import Styles from '../../../styles/Style'

export default class AddCallButton extends Component {
  static propTypes = {
    /** If true, will show the button label */
    showLabel: PropTypes.bool,
    /** Caregiver button can be disabled */
    enableCaregivers: PropTypes.bool,
    /** Interpreters can be disabled */
    enableInterpreters: PropTypes.bool,
    /** Called to add a staff member */
    onAddStaff: PropTypes.func.isRequired,
    /** Called after the caregivers button is clicked */
    onAddCaregiver: PropTypes.func.isRequired,
    /** Called to add an intepreter */
    onAddInterpreter: PropTypes.func.isRequired,
    /** Class for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
    /** Indicating if more participants can be added to the call */
    canAddMoreParticipants: PropTypes.bool.isRequired,
  }
  static defaultProps = {
    showLabel: true,
    enableCaregivers: true,
    className: '',
    style: {},
  }

  state = {
    showMenu: false,
    showAlert: false,
  }

  toggleMenu = () => {
    if (this.props.canAddMoreParticipants) {
      this.setState({ showMenu: !this.state.showMenu })
    } else {
      this.setState({ showAlert: !this.state.showMenu })
    }
  }

  handleMenuClick = item => {
    this.setState({ showMenu: false })

    if (item === 'staff') this.props.onAddStaff()
    if (item === 'caregiver') this.props.onAddCaregiver()
    if (item === 'interpreter') this.props.onAddInterpreter()
  }

  handleClose = () => {
    this.setState({ showAlert: false })
  }

  render() {
    const { showLabel, enableCaregivers, enableInterpreters, style, className } = this.props
    const { showMenu, showAlert } = this.state

    const label = showLabel ? <CallButtonLabel text="Add Call" /> : null

    return (
      <div className={`addcallbutton-wrap ${className}`} style={style}>
        <AddCallMenu
          show={showMenu}
          enableCaregivers={enableCaregivers}
          enableInterpreters={enableInterpreters}
          onAddStaff={() => {
            this.handleMenuClick('staff')
          }}
          onAddCaregiver={() => {
            this.handleMenuClick('caregiver')
          }}
          onAddInterpreter={() => {
            this.handleMenuClick('interpreter')
          }}
          onBack={this.toggleMenu}
        />
        <ModalToast show={showAlert} data-testid="maximum-reached-modal">
          <div className="modalalert">
            <ModalToast.Header
              className="modalalert"
              title="'Sorry, the call is currently full. You can add someone after a caller has left'"
            />
            <ModalToast.Button fluid onClick={this.handleClose} />
          </div>
        </ModalToast>
        <div className={`addcallbutton-button `}>
          <CallButton
            color={Styles.brandcolorDark}
            name="add-call"
            ghost={true}
            ghostOpacity={0.6}
            onClick={this.toggleMenu}
            data-testid="button-add-call">
            <IconPlus />
          </CallButton>

          {label}
        </div>
      </div>
    )
  }
}
