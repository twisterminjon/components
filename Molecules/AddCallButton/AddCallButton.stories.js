import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, boolean } from '@storybook/addon-knobs'

import AddCallButton from './AddCallButton'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('AddCallButton', () => {
  const showLabelLabel = 'showLabel'
  const showLabelDefault = true
  const showLabelValue = boolean(showLabelLabel, showLabelDefault)
  const showMaxReached = boolean('showMaxReached', true)

  return (
    <div
      style={{
        position: 'fixed',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        backgroundColor: 'violet',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      }}>
      <AddCallButton
        showLabel={showLabelValue}
        enableCaregivers={true}
        enableInterpreters={true}
        onAddStaff={() => alert('adding staff')}
        onAddCaregiver={() => alert('adding caregiver')}
        onAddInterpreter={() => alert('adding interpreter')}
        canAddMoreParticipants={showMaxReached}
      />
    </div>
  )
})
