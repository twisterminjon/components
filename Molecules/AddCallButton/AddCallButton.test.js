import React from 'react'
import AddCallButton from './AddCallButton'

describe('AddCallButton', () => {
  const mockFun = jest.fn()

  it('matches the snapshot with a label', () => {
    const wrapper = window.shallow(
      <AddCallButton
        showLabel={true}
        enableCaregiver={true}
        enableInterpreter={true}
        onAddStaff={mockFun}
        onAddCaregiver={mockFun}
        onAddInterpreter={mockFun}
        canAddMoreParticipants={true}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot with out a label', () => {
    const wrapper = window.shallow(
      <AddCallButton
        showLabel={false}
        enableCaregiver={false}
        enableInterpreter={true}
        onAddStaff={mockFun}
        onAddCaregiver={mockFun}
        onAddInterpreter={mockFun}
        canAddMoreParticipants={true}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('shows the menu when clicked', () => {
    const wrapper = window.mount(
      <AddCallButton
        showLabel={false}
        enableCaregiver={false}
        enableInterpreter={true}
        onAddStaff={mockFun}
        onAddCaregiver={mockFun}
        onAddInterpreter={mockFun}
        canAddMoreParticipants={true}
      />
    )
    // menu should be hidden to start
    expect(
      wrapper
        .find('.modaltoast-wrap')
        .find('[data-testid="add-to-call-modal"]')
        .hasClass('modaltoast--show')
    ).toBeFalsy()

    // find and click the button
    const buttonContainer = wrapper.find('[data-testid="button-add-call"]')
    const button = buttonContainer.find('button')
    button.simulate('click')

    // menu should be hidden to start
    expect(
      wrapper
        .find('[data-testid="add-to-call-modal"]')
        .find('.modaltoast-wrap')
        .hasClass('modaltoast--show')
    ).toBeTruthy()
  })

  it('shows the alert when clicked', () => {
    const wrapper = window.mount(
      <AddCallButton
        showLabel={false}
        enableCaregiver={false}
        enableInterpreter={true}
        onAddStaff={mockFun}
        onAddCaregiver={mockFun}
        onAddInterpreter={mockFun}
        canAddMoreParticipants={false}
      />
    )
    // menu should be hidden to start
    expect(
      wrapper
        .find('[data-testid="maximum-reached-modal"]')
        .find('.modaltoast-wrap')
        .hasClass('modaltoast--show')
    ).toBeFalsy()

    // find and click the button
    const buttonContainer = wrapper.find('[data-testid="button-add-call"]')
    const button = buttonContainer.find('button')
    button.simulate('click')

    // menu should be hidden to start
    expect(
      wrapper
        .find('[data-testid="maximum-reached-modal"]')
        .find('.modaltoast-wrap')
        .hasClass('modaltoast--show')
    ).toBeTruthy()
  })
})
