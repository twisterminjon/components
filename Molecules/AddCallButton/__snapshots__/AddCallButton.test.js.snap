// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCallButton matches the snapshot with a label 1`] = `
<div
  className="addcallbutton-wrap "
  style={Object {}}
>
  <AddCallMenu
    enableCaregivers={true}
    enableInterpreters={false}
    onAddCaregiver={[Function]}
    onAddInterpreter={[Function]}
    onAddStaff={[Function]}
    onBack={[Function]}
    show={false}
  />
  <ModalToast
    data-testid="maximum-reached-modal"
    show={false}
  >
    <div
      className="modalalert"
    >
      <ModalToastHeader
        className="modalalert"
        title="'Sorry, the call is currently full. You can add someone after a caller has left'"
      />
      <ModalToastButton
        fluid={true}
        onClick={[Function]}
      />
    </div>
  </ModalToast>
  <div
    className="addcallbutton-button "
  >
    <CallButton
      color="#002862"
      data-testid="button-add-call"
      dimmed={false}
      disabled={false}
      ghost={true}
      ghostOpacity={0.6}
      name="add-call"
      onClick={[Function]}
      size="medium"
    >
      <IconPlus
        className=""
        color="black"
        size={25}
        style={Object {}}
      />
    </CallButton>
    <CallButtonLabel
      text="Add Call"
    />
  </div>
</div>
`;

exports[`AddCallButton matches the snapshot with out a label 1`] = `
<div
  className="addcallbutton-wrap "
  style={Object {}}
>
  <AddCallMenu
    enableCaregivers={true}
    enableInterpreters={false}
    onAddCaregiver={[Function]}
    onAddInterpreter={[Function]}
    onAddStaff={[Function]}
    onBack={[Function]}
    show={false}
  />
  <ModalToast
    data-testid="maximum-reached-modal"
    show={false}
  >
    <div
      className="modalalert"
    >
      <ModalToastHeader
        className="modalalert"
        title="'Sorry, the call is currently full. You can add someone after a caller has left'"
      />
      <ModalToastButton
        fluid={true}
        onClick={[Function]}
      />
    </div>
  </ModalToast>
  <div
    className="addcallbutton-button "
  >
    <CallButton
      color="#002862"
      data-testid="button-add-call"
      dimmed={false}
      disabled={false}
      ghost={true}
      ghostOpacity={0.6}
      name="add-call"
      onClick={[Function]}
      size="medium"
    >
      <IconPlus
        className=""
        color="black"
        size={25}
        style={Object {}}
      />
    </CallButton>
  </div>
</div>
`;
